src/basic/fd-util.c

```diff
@@ -913,6 +913,36 @@ int fd_is_opath(int fd) {
         return FLAGS_SET(r, O_PATH);
 }
 
+int fd_extrinsic_flags_safe(int fd) {
+        unsigned unexpected_flags;
+        int flags;
```

Comment:
hmm, so fnctl() returns int and open() takes an int for these flags. i always find it weird to have signed flags params, but it is how it is, hence why deviate from it for `unexpected_flags`? i'd just make it int too. just masking out some flags shouldn#t change the type of the flags field...

Reply:
Hmm, so `%o` only accepts unsigned int. Since we're doing casting anyway I think we may as well do it here...

Reply:
hmm, why octal btw? that's weird, exept for access modes noone ever uses octal?

use %x instead?

and just cast the param when passing it to printf()?